### PR TITLE
feat(framework): Control logging

### PR DESCRIPTION
**Pull request by SokratisVidros**
### What changed? Why was the change needed?

Add coarse-graining logging control and test logs to prevent flooding production. 

This is the first step in switching on/off framework logging so we can use it in production and e2e tests. In future work, we need to add logging levels and log differently in development (log everything) and production (log only errors and a summary of the discovered workflows).

### CODE REVIEW

File: packages/framework/src/handler.ts [hunk 1]
```diff
@@ -275,17 +275,17 @@ export class NovuRequestHandler<Input extends any[] = any[], Output = any> {
...
 
       return this.createError(error);
     } else if (isPlatformError(error)) {
       return this.createError(error);
     } else {
       const bridgeError = new BridgeError(error);
-      // eslint-disable-next-line no-console
-      console.error(bridgeError);
+      // eslint-disable-next-line @typescript-eslint/no-base-to-string
+      log((l) => l.error(bridgeError.message || bridgeError.toString()));
```
Comment (djabarovgeorge) [inline packages/framework/src/handler.ts:288]: feels a bit complex and not the usual format of logging. what are the cons of doing something like this?


```suggestion
      log.error(bridgeError.message || bridgeError.toString());
```



in case of 
```
    log((l) => `${prefix} ${l.emoji.STEP} Discovered stepId: '${step.stepId}'\tType: '${step.type}'`);
```
we could do 
```
    log.info(`${prefix} ${log.emoji.STEP} Discovered stepId: '${step.stepId}'\tType: '${step.type}'`);
```

the formatting should be the responsibility of the logger user; this way, it is easier to read. in addition, having the default log level hidden is a bit confusing and not straightforward.

File: packages/framework/src/client.ts [hunk 2]
```diff
@@ -72,21 +73,31 @@ export class Client {
 
   public strictAuthentication: boolean;
 
+  public logging: boolean = true;
```
Comment (djabarovgeorge) [inline packages/framework/src/client.ts:76]:
```suggestion
  public loggingEnabled | isLoggingEnabled: boolean = true;
```

File: packages/framework/src/utils/log.utils.ts [hunk 3]
```diff
@@ -1,29 +1,78 @@
...
+    // eslint-disable-next-line no-console
+    console.log(message(f));
+  }
+}
+
+log.enable = () => {
+  enabled = true;
+};
+log.disable = () => {
+  enabled = false;
```
Comment (djabarovgeorge) [inline packages/framework/src/utils/log.utils.ts:77]: 👏

File: packages/framework/src/utils/options.utils.ts [hunk 4]
```diff
@@ -5,3 +5,15 @@ export function resolveApiUrl(providedApiUrl?: string): string {
...
+export function resolveLogging(providedLogging?: boolean): boolean {
+  if (providedLogging !== undefined) {
+    return providedLogging;
+  }
+  if (process.env.NOVU_LOGGING !== undefined) {
+    return process.env.NOVU_LOGGING === 'true';
+  }
+
+  // Disable verbose logging in test and production environments
+  return !['test', 'production'].includes(process.env.NODE_ENV);
```
Comment (djabarovgeorge) [inline packages/framework/src/utils/options.utils.ts:18]: Love it! 
But can't we reuse NOVU_LOGGING to toggle this? so for test and prod it will be false, but in case we want to debug some edge cases, we could turn it on in prod. 
in addition, we could use more specific env variables so they won't collide in the future in any way FRAMEWORK_LOGGING, although it should not be an issue.
